onnxruntime/core/providers/cuda/cuda_graph.cc

```diff
@@ -30,6 +40,29 @@ void CUDAGraph::CaptureBegin() {
 }
 
 void CUDAGraph::CaptureEnd() {
+  if (!IsGraphCaptureAllowedOnRun()) {
+    return;
+  }
+
+  if (cuda_graph_annotation_id_.has_value()) {
+    CUDA_CALL_THROW(cudaStreamEndCapture(stream_, &additional_graph_));
+    if (additional_graph_ == NULL) {
+      ORT_THROW("CUDAGraph::CaptureEnd: additional_graph_ is NULL");
+    }
+
+    cudaGraphExec_t graph_exec = NULL;
+
+    has_additional_graph_ = true;
+    CUDA_CALL_THROW(cudaGraphInstantiate(&graph_exec, additional_graph_, NULL, NULL, 0));
+    CUDA_CALL_THROW(cudaGraphDestroy(additional_graph_));
+    has_additional_graph_ = false;
+
+    GraphAnnotation_t cuda_graph_id = cuda_graph_annotation_id_.value();
+    graph_exec_map_.emplace(cuda_graph_id, graph_exec);
```

Comment:
What is the expected behavior if someone tries to re-capture a new graph for an annotation_id that has already been used ? `emplace()` will do nothing - not even replace the already existing key. This may lead to a lot of ambiguity.

Reply:
the design currently only do replay with that id

Reply:
Ok. Sounds like capturing another graph for same annotation id is not supported. Maybe we need to add some log/throw exception if user accidentally re-use an annotation id ?

Reply:
If we want to re-capture, the design need two run options like capture_graph_id and replay_graph_id.

Reply:
Ahh yes. I forgot that the scenario is - given an annotation id, capture a graph if it doesn't exist. if it does exist, replay the graph.